flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -409,13 +409,30 @@ public static RpcService createRpcService(
 		if (taskManagerHostname != null) {
 			LOG.info("Using configured hostname/address for TaskManager: {}.", taskManagerHostname);
 		} else {
-			Time lookupTimeout = Time.milliseconds(AkkaUtils.getLookupTimeout(configuration).toMillis());
-
-			InetAddress taskManagerAddress = LeaderRetrievalUtils.findConnectingAddress(
-				haServices.getResourceManagerLeaderRetriever(),
-				lookupTimeout);
-
-			taskManagerHostname = taskManagerAddress.getHostName();
+			InetAddress taskManagerAddress;
+
+			String bindPolicy = configuration.getString(TaskManagerOptions.HOST_BIND_POLICY);
+			switch (bindPolicy.toLowerCase()) {
+				case "hostname":
```

Comment:
could be an enum. We could then also use enum to generate the description for `HOST_BIND_POLICY`.